### ISSUE TITLE
Scale annotations are leaking

### ISSUE THREAD

**Issue by hvelarde** [collective/collective.cover#828]
I was checking annotations on one cover object and I found the following:

```pypdb
>>> len([k for k in IAnnotations(obj) if k.startswith('plone.tiles.data')])
85
>>> len([k for k in IAnnotations(obj) if k.startswith('plone.tiles.configuration')])
61
>>> len([k for k in IAnnotations(obj) if k.startswith('plone.tiles.scale')])
200
```

that means we have annotations for 200 scales but only 28 of them seem to be related with existing tiles:

```pypdb
>>> scales = {k.split('.')[3] for k in IAnnotations(obj) if k.startswith('plone.tiles.scale')}
>>> tiles = {k.split('.')[3] for k in IAnnotations(obj) if k.startswith('plone.tiles.data')}
>>> len(scales & tiles)
28
```

**Comment by hvelarde** [comment]
I did the following test using [collective.revisionmanager](https://github.com/collective/collective.revisionmanager):

* calculate statistics and getting information about the object
* purge all but 5 revisions
* recalculate statistics

object size is reduced, but we have the same number of annotations present; this seems to me as an issue with versioning.

![before](https://user-images.githubusercontent.com/341393/45656901-59538b00-babe-11e8-913c-c940792e61cd.png)
![after](https://user-images.githubusercontent.com/341393/45656900-59538b00-babe-11e8-94c1-ddf514f9fe8b.png)

**Comment by rafaelbco** [comment]
The code in the first comment only gets the annotations for the current object. After erasing revisions the object and it's annotations are expected to remain intact. Does that makes sense or am I missing some detail?

**Comment by idgserpro** [comment]
This is just another example of why https://github.com/collective/collective.cover/issues/742 is not that trivial...

**Comment by hvelarde** [comment]
that's right; I think the problem is versioning: versions store their data in annotations on the same base object and, when they are removed, annotations remain in place.

**Comment by idgserpro** [comment]
Since we don't have a properly CMFEditions configuration, is https://github.com/collective/collective.cover/blob/f66adb50d2834301ff7023a4f1bfe23bd0438039/src/collective/cover/profiles/default/repositorytool.xml really necessary? Do you think it may be the reason we're having this leakage?

**Comment by hvelarde** [comment]
I have no idea :)

**Comment by rafaelbco** [comment]
I think that image scales should never be stored in the DB. They should be cached in the RAM cache and be regenerated when need, ie. when cached scales expires. 

I had a look at the scaling code and find a bit confusing. I was not able to implement this idea without changing code in more than one package, so I gave up. I was hoping I could do it with ZCA, but nope. 

I wrote a modifier for cmfeditions that prevents scales from being versioned. Does that helps you guys?

**Comment by rafaelbco** [comment]
I have an idea, which is only tangentially related to this issue, but it would solve it if 
implemented.

I think setups where scales are heavily used (think site front pages) and where a  cache frontend is used (Varnish), image scales could be stored only in a RAM cache, without losing much performance. 

In a RAM cache things expire and are evicted from the cache. So the scales storage should be able to rebuild a requested scale when needed.

I have an idea about how to implement it. If you like this approach I'd appreciate feedback
on this: plone/plone.scale#32
